[Tan Jin Waye] iP

Tan-Jin-Waye opened the pull request:
> "This is supposed to be a quote. I'll use this space instead to be really annoying" ~Me

# Duke


## Features
- View your list of tasks (most of them unfinished probably)
- Add **even more** tasks <sub> yay </sub>
- ~~Check them off~~ probably not 

Types of fun tasks!
1. Todos are *basic*
2. Deadlines have a best by date
3. Events you will probably miss anyway


## How 2 Use
~~~java
//run me !
public class Launcher {  
    public static void main(String[] args) {
        Application.launch(Main.class, args); 
    }
}

~~~

## Tracking My Own Progress
- [x] Week 1
- [x] Week 2
- [ ] Uhh-


## 😄 😄 😄 
[Look at more embarrassing code here!](https://github.com/Tan-Jin-Waye/ip)

Commit:
Add Gradle support

Tan-Jin-Waye committed:
Updated Duke.java for functionality: Greet, Echo, Exit.

Tan-Jin-Waye committed:
Updated Duke.java for functionality: Add, List.

Tan-Jin-Waye committed:
Duke.java: Refactored handler() into smaller methods.

Tan-Jin-Waye committed:
Task.java: New class.
Duke.java: Edited code to work with new Task class.

Tan-Jin-Waye committed:
Updated Duke.java, Task.java. Added Todo.java, Deadline.java, Event.java.

Tan-Jin-Waye committed:
Duke.java: Refactored code to run iteratively, and create a Duke instance instead of static.

Tan-Jin-Waye committed:
Duke.java: updated code to accept multiple lines of input at once. text-ui-test: Updated input.txt and EXPECTED.txt with some test cases.

Tan-Jin-Waye committed:
Added DukeException.java and some subclasses.

Tan-Jin-Waye committed:
Updated messages of DukeException.java and subclasses.

Tan-Jin-Waye committed:
Duke.java: Handled exceptions as per Level-5 req.

Tan-Jin-Waye committed:
Duke.java: Added functionality for remove as per Level-6. Exceptions handled.

Tan-Jin-Waye committed:
Updated text-ui-test with more test cases, updated expected result accordingly.

Tan-Jin-Waye committed:
Add SaveManager.java to handle reading and writing list to file

Tan-Jin-Waye committed:
Modify Read() access modifier from private to public

Tan-Jin-Waye committed:
Modify Task abstract class to implement serializeable

Tan-Jin-Waye committed:
Modify Duke.java to work with SaveManager.java

Tan-Jin-Waye committed:
Update .gitignore to account for save file

Tan-Jin-Waye committed:
Update Deadline.java, Event.java to store date in LocalDate object

Tan-Jin-Waye committed:
Update regex for Deadline and Event in Duke.java

Tan-Jin-Waye committed:
Handle exception relating to incorrect date format

Tan-Jin-Waye committed:
Merge branch 'branch-Level-7'

Tan-Jin-Waye committed:
Merge branch 'branch-Level-8'

Tan-Jin-Waye committed:
Rectify syntax error derived from merging two branches into master

Tan-Jin-Waye committed:
Correct formatting error when printing out task items

Tan-Jin-Waye committed:
Refactor SaveManager into Storage

Tan-Jin-Waye committed:
Refactor UI components into Ui.java and edited affected files

Tan-Jin-Waye committed:
Refactor Task Array into TaskList and update Duke.java accordingly

Tan-Jin-Waye committed:
Rename function names of Storage and updated affected classes accordingly

Tan-Jin-Waye committed:
Refactor operations to Parser.java and edited affected classes appropriately

Tan-Jin-Waye committed:
Organise all classes into package duke, add Main method

Tan-Jin-Waye committed:
Organise DukeException and subclasses into exceptions package and updated affected classes accordingly

Tan-Jin-Waye committed:
Merge remote-tracking branch 'nus-cs2103-AY2223S1/add-gradle-support'

Tan-Jin-Waye committed:
Change mainClassName to be able to run Duke in Gradle

Tan-Jin-Waye committed:
Add JUnit testing for following classes: Deadline, Event, Todo

Tan-Jin-Waye committed:
Add appropriate Javadocs comments to relevant classes except Main and Exceptions

Tan-Jin-Waye committed:
Complete Level-9 by adding methods to TaskList and Ui, update Parser methods accordingly

Tan-Jin-Waye committed:
Edit code to fit Java Coding Standards

Tan-Jin-Waye committed:
Merge branch 'branch-Level-9'

Tan-Jin-Waye committed:
Merge branch 'branch-A-Javadoc'

Tan-Jin-Waye committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/TaskList.java

Tan-Jin-Waye committed:
Edit Storage, Ui to be called from instanced context, not static

Tan-Jin-Waye committed:
Replace wildcard imports

Tan-Jin-Waye committed:
Configure checkstyles and edit code based on checkstyle results

Tan-Jin-Waye committed:
Merge branch 'A-Checkstyle'

Tan-Jin-Waye committed:
Include Gui class to test JavaFX

Tan-Jin-Waye committed:
Revert "Include Gui class to test JavaFX"

This reverts commit 5508976e32528d8daa7cc88263c1de084adc0a9c.

Tan-Jin-Waye committed:
Add Gui class for JavaFX testing

Tan-Jin-Waye committed:
Update Gui with more components and styling

Tan-Jin-Waye committed:
Add rudimentary user interaction to Duke Gui

Tan-Jin-Waye committed:
Edit files to use FXML

Tan-Jin-Waye committed:
Convert Duke to FXML Gui

Tan-Jin-Waye committed:
Edit files to properly return response from user input

Tan-Jin-Waye committed:
Merge branch 'branch-Level-10'

jontmy commented on build.gradle:
```diff
@@ -0,0 +1,67 @@
+//javafx {
+//    version = "18.0.2"
+//    modules = ['javafx.controls', 'javafx.fxml']
+//}
```
Perhaps rely on Git to restore these lines (instead of commenting them out) if they're needed in future?

jontmy commented on src/main/java/Launcher.java:
```diff
@@ -0,0 +1,9 @@
+
+public class Launcher {
+    public static void main(String[] args) {
+        //Duke.main(args);
```
Should this line be deleted instead?

jontmy commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,117 @@
+     * @return boolean end to determine execution finish
+     * @throws DukeException when runtime errors occur
+     */
+    public String handler(String input) throws DukeException {
```
Would naming this method `handle` as a verb be more appropriate?

jontmy commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,117 @@
+     */
+    public String handler(String input) throws DukeException {
+        String[] args = input.split(" ", 2);
+        //boolean isEnded = false;
```
There are a number of lines commented-out in this class relating to `isEnded` - perhaps delete them and rely on Git to restore them if needed in future?

jontmy commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,41 @@
+    /**
+     * Toggles the completed variable between true/false
+     */
+    public void completeToggle() {
```
Would `toggleComplete` be a more suitable name, as it would start with the verb?

jontmy commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,87 @@
+import duke.exceptions.DukeListOobException;
+import duke.exceptions.DukeWrongInputException;
+
+public class TaskList {
+
+    private ArrayList<Task> tasks;
```
Considering that the methods will be invoked on a `TaskList`, I think that prefixing `list` to each of the methods in this class is redundant, and potentially misleading in the case of `listDelete` which suggests that it deletes an entire list.

jontmy commented on src/main/java/duke/exceptions/DukeException.java:
```diff
@@ -0,0 +1,7 @@
+package duke.exceptions;
+
+public abstract class DukeException extends Exception {
+    public DukeException(String msg) {
+        super(msg);
+    }
+}
```
I like how you've created multiple exceptions that extend from `DukeException`, and also placed them in their own sub-package. Nice!

jontmy commented on .gitignore:
```diff
@@ -15,3 +15,6 @@ bin/
 text-ui-test/EXPECTED-UNIX.TXT
+
+SaveData.txt
+/src/main/resources/images/*.png
```
Hi! I cloned your project, but was unable to run it because these images have been gitignore'd,
meaning the application throws `NullPointerExceptions` as the images can't be found and loaded.

jontmy commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,50 @@
+                }
+                System.out.println(response);
+            } catch (DukeException e) {
+                System.out.println(ui.printException(e));
```
Seems a little quirky that the method named `printException` has to be wrapped inside a `println` - perhaps the UI can handle the printing directly?

jontmy commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,41 @@
+package duke;
+
+import java.io.Serializable;
+public abstract class Task implements Serializable {
```
Nice use of Java's `Serializable` interface to read/write the tasks to a file!

I wanted to test your implementation of this, but unfortunately the issue caused by the gitignore'd images mentioned above meant that I couldn't.

jontmy reviewed (commented):
Great work! Your code is pleasant to read, and there are some neat Java features that were used (such as the `Serializable` interface)!

I would suggest taking a look at the commented-out lines and some of the method names though, as the coding quality standard recommends that method names should start with a verb.

Nonetheless, your code is well organized and logically structured, which made reviewing the code quality a breeze!

Tan-Jin-Waye committed:
Add assertions to Parser and Storage

No assertions present in code.

Storage identified to be one point of failure as reading text file can
incur a file not found error, and file contents use unchecked typecast
to fit tasks arraylist.
Parser identified to be another point of failure as users can enter
blank strings into task description.

Lets add assertions to document assumptions regarding abovementioned
points of failure.

Tan-Jin-Waye committed:
Make minor Coding Standard adjustments

Many Coding Standard violations present.

As a baseline for Code Quality, it is important that these get
addressed first.

Let's address Coding Standard violations here.

Tan-Jin-Waye committed:
Make edits for Code Quality

Tan-Jin-Waye committed:
Merge pull request #1 from Tan-Jin-Waye/branch-A-CodeQuality

Merge branch 'A-CodeQuality'

Tan-Jin-Waye committed:
Merge branch 'master' into branch-A-Assertions

# Conflicts:
#	src/main/java/duke/Parser.java

Tan-Jin-Waye committed:
Merge pull request #2 from Tan-Jin-Waye/branch-A-Assertions

Add assertions to Parser and Storage

Tan-Jin-Waye committed:
Set up for CI

Tan-Jin-Waye committed:
Merge branch 'branch-A-CI'

Tan-Jin-Waye committed:
Implement remind command

Remind command not implemented.

Reminders is a useful extension to have and also relevant to
tP.

Let's implement reminders.

Tan-Jin-Waye committed:
Merge pull request #3 from Tan-Jin-Waye/branch-B-Reminders

Implement remind command

Tan-Jin-Waye committed:
Edit code in line with checkstyle

Tan-Jin-Waye committed:
Remove checkstyle from build.gradle

Tan-Jin-Waye committed:
Add exception handling for empty descriptions

Tan-Jin-Waye committed:
Suppress unchecked cast warning in storage

Tan-Jin-Waye committed:
Merge branch 'branch-A-MoreErrorHandling'

Tan-Jin-Waye committed:
Add testing for TaskList and Ui

Tan-Jin-Waye committed:
Merge branch 'branch-A-MoreTesting'

Tan-Jin-Waye committed:
Edit DialogBox with some simple styling

Tan-Jin-Waye committed:
Merge branch 'branch-A-BetterGui'

Tan-Jin-Waye committed:
Make some edits to Storage and SaveData.txt

Tan-Jin-Waye committed:
Add product screenshot and update README

Tan-Jin-Waye committed:
Fix a bug concerning reminds command

Tan-Jin-Waye committed:
Edit code in line with checkstyle

Tan-Jin-Waye committed:
Update README

Tan-Jin-Waye committed:
Make minor edits for release

Tan-Jin-Waye committed:
Update README.md

Tan-Jin-Waye committed:
Update README.md
